app/channel.go

```diff
@@ -585,11 +585,7 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 }
 
 func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
-	result := <-a.Srv.Store.Channel().Restore(channel.Id, model.GetMillis())
```

Comment:
I would expect here to maintain the previous approach, `Restore` returns the error and if not, return the channel received as parameter.